### DISCUSSION TITLE
Webpack-dev-server help: browser shows directories instead of rendering website

### DISCUSSION THREAD

**Discussion by wes-davis** [webpack/webpack#14819] [Q&A]
Hey all. 

I am trying to add webpack-dev-server to my local server environment for a Wordpress custom theme project. 

Right now I've successfully installed and can run WDS, but when I open the host in a browser, it just shows a folder directory instead of the actual website homepage. 

No matter where in the project directory tree I point `static`, it still just opens a *folder view* at that level in the browser — can anyone suggest what I might be missing/doing wrong config-wise?

Added to webpack.config.js:
```
devServer: {
        host: 'xxxxxxxxxxxxx.local',            // my Local site—but the same thing happens if I just use 'localhost'
        static: path.join(__dirname, './'),    // I've already tried changing this to various other levels of the folder structure
        https: true,
        hot: true,
        port: 3000
    }
```

**Comment by alexander-akait** [comment]
It means you don't have `index.html`

**Reply by wes-davis** [reply]
Makes sense — a Wordpress theme uses `index.php` instead. 

So is there maybe a rule/setting I can add to the config that would tell WDS to look for `index.php` instead of `index.html`?

**Reply by alexander-akait** [reply]
But `index.php` is just a file, you need to run php dev server and proxy requests to php dev server (but not proxy webpack files)

**Reply by wes-davis** [reply]
> run php dev server and proxy requests to php dev server (but not proxy webpack files) 

I keep trying and failing to parse this sentence… I'm really new to webpack — could you elaborate a bit, or point me to anywhere I might be able to find reliable details on how to integrate webpack-dev-server with the php/Wordpress environment? 

Overall I have webpack working fine so far to handle `watch`, PostCSS etc — I really just want to be able to get Hot Module Replacement working too.

**Reply by alexander-akait** [reply]
Old, but may help https://by-robots.dev/2019/01/webpack-and-wordpress/ (some things are deprecated), I recommend to use stackowerflow or ask it in wordpress community.

Shorty - you should run `wordpress` using php dev server, here CLI for it https://developer.wordpress.org/cli/commands/server/, when you need to configure webpack dev server to proxy your request (https://webpack.js.org/configuration/dev-server/#devserverproxy, note - you need don't need proxy webpack files, you can store them in separate directory, look at `output.path`), then run webpack dev server (i.e. `webpack serve`), after this you can open browser and logic will be:

browser -> request -> webpack dev server -> it is request to webpack generated files? 
___________________________________________________-> yes -> dev server handle it and return to browser
___________________________________________________-> no -> proxy to php server -> run php code -> html -> return to browser
